frontend/packages/noobaa-storage-plugin/src/components/capacity-card/capacity-card-body.tsx

```diff
@@ -1,106 +1,96 @@
-    total += tmpMetricData.y;
-    metricData.push(tmpMetricData);
-  });
-  metricData = _.sortBy(metricData, ['y']).reverse();
-  // collect the other data
-  const extraData = metricData.splice(maxSize, metricData.length);
-  // if there are any extra / other items, reduce it to a single entry
-  if (extraData.length) {
-    const otherData = extraData.reduce((prev, curr) => {
-      return { x: 'Others', y: prev.y + curr.y };
-    });
-    metricData.push(otherData);
+  // if there is no data, return the empty values
+  if (!metricData || !metricData.length) {
+    return { metricData, metricLegend };
   }
-  metricData = metricData.map((m) => {
-    metricLegend.push({ name: m.x });
-    return { x: m.x, y: (m.y / total) * 100 };
+  // get the percentage
+  metricData.forEach((m) => {
+    metricLegend.push({ name: `${m.x}` });
+    m.y = total ? (m.y / total) * 100 : 100;
   });
-  return { metricData, metricLegend, total };
+  return { metricData, metricLegend };
 };
 
-export const CapacityCardBody: React.FC<CapacityCardBodyProps> = ({ isLoading, metricsData }) => {
+export const CapacityCardBody: React.FC<CapacityCardBodyProps> = ({
+  isLoading,
+  metricsData,
+  totalUsage,
+}) => {
   if (isLoading) {
     return <LoadingInline />;
   }
-  // trying to avoid redundant code
-  const { metricData, metricLegend, total } = getMetricDataAndLegend(metricsData);
-  if (!metricData.length) {
+  const { metricData, metricLegend } = getLegendAndChartData(metricsData, totalUsage);
+  if (!metricData || !metricData.length) {
     return (
-      <EmptyState className="chart-empty-state" variant={EmptyStateVariant.full}>
-        <EmptyStateIcon size="sm" icon={ChartPieIcon} />
-        <Title size="sm">No Prometheus datapoints found</Title>
-      </EmptyState>
+      <div className="nb-capacity-card__chart-no-data donut-utilization-chart-legend-right">
+        <ChartDonutUtilization
+          labels={['']}
+          data={{ y: 0 }}
+          invert
+          subTitle={null}
+          themeColor={ChartThemeColor.green}
+          title="No data"
+        />
+      </div>
     );
   }
+  const totalHumanized = humanizeBinaryBytesWithoutB(totalUsage, null, totalUsage ? null : 'Mi');
   return (
-    <div>
-      <span className="text-secondary">
-        <h4>Usage Breakdown</h4>
-      </span>
-      <ChartDonut
-        donutHeight={250}
-        donutWidth={110}
-        labelPosition="startAngle"
-        legendPosition="right"
-        legendData={metricLegend}
-        legendOrientation="vertical"
-        data={metricData}
-        labels={(datum) => `${datum.x}: ${datum.y.toFixed(2)}%`}
-        subTitlePosition="bottom"
-        themeColor={ChartThemeColor.multi}
-        themeVariant={ChartThemeVariant.light}
-        title={humanizeBinaryBytesWithoutB(total).string}
-        radius={53}
-        innerRadius={46}
-      />
+    <div className="nb-capacity-card__chart-body">
+      <div className="noobaa-capacity-card__item">
```

Comment:
why the nested div ? also `noobaa-capacity-card__item` does not exist

Reply:
ack